### PR TITLE
Overwatch Signals for QDEL: Less client disconnection?

### LINKED ISSUE #3444: Disconnections from server due to xeno eyeing a xeno before they gib
### Testmerges

N/A

### Description of the bug

If you xeno eye a xeno, and they gib (like in an acid runner explosion), you will disconnect from the server for a bit.

### What's the difference with what should have happened?

Getting disconnected can mean life or death. Ideally, there would be no disconnect here.

### How do we reproduce this bug?

1. Be xeno (not queen)
2. Eye (watch xenomorph) an acid runner while they are exploding 
3. When the acid runner explodes, your client will disconnect from server
...
//
This is pretty consistently reported from what i know, for a lot of people it WILL disconnect you if you eye an acid runner while they are gibbing.
Not sure if it's exclusively acid rouny gibbing though. 

### Issue Bingo

- [ ] Issue could be reproduced at least once
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

### PR DESCRIPTION
**Pull request by Drulikar**
# About the pull request

This PR is sort of a stab in the dark to fix #3444 by making marine and xeno overwatching register a signal for qdeletion to end the overwatch faster. The actual source of this issue is apparently that the multithreaded map tick beta has an issue with crashing clients, so the changes made in this PR don't need to be permanent if the effect isn't as desirable when the beta is fixed.

Locally testing has not been able to reproduce the problem, but I can confirm I do not get the momentary blank screen when overwatching something that gets deleted.

# Explain why it's good for the game

Hopefully no more (or less) client disconnections when overwatching something that gets deleted.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

qdel

</details>

# Changelog
:cl: Drathek
fix: Clients should hopefully avoid disconnection when observing someone that is deleted.
/:cl:

### CODE REVIEW

**Comment by Wintermote** [comment]
I haven't noticed disconnection when overwatching using the overwatch console in CIC, and that should be the same process as the ghost overwatch, shouldn't it?

File: code/modules/mob/living/carbon/xenomorph/XenoOverwatch.dm [hunk 1]
```diff
@@ -109,15 +110,22 @@
...
 		observed_xeno.hud_set_queen_overwatch()
 		SEND_SIGNAL(src, COMSIG_XENO_OVERWATCH_XENO, observed_xeno)
 		RegisterSignal(src, COMSIG_MOB_MOVE_OR_LOOK, PROC_REF(overwatch_handle_mob_move_or_look))
+		RegisterSignal(observed_xeno, COMSIG_PARENT_QDELETING, PROC_REF(overwatch_handle_deletion))
 
 	src.reset_view()
 
+/mob/living/carbon/xenomorph/proc/overwatch_handle_deletion(mob/living/carbon/xenomorph/deleted, forced)
+	SIGNAL_HANDLER
+	UnregisterSignal(deleted, COMSIG_PARENT_QDELETING)
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/XenoOverwatch.dm:126]: this isn't necessary

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -356,6 +356,10 @@
 		SEND_SIGNAL(client, COMSIG_CLIENT_RESET_VIEW, A)
 	return
 
+/mob/proc/reset_observer_view_on_deletion(atom/deleted, force)
+	SIGNAL_HANDLER
+	UnregisterSignal(deleted, COMSIG_PARENT_QDELETING)
```
Comment (Watermelon914) [inline code/modules/mob/mob.dm:361]: this isn't necessary, all signals get unregistered on deletion anyways

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]
lets see

**Comment by Drulikar** [comment]
#3525 might be a more comprehensive fix unless that's yet another different 515 flaw.

**Pull request merged**